### ISSUE TITLE
Linking nonexistent files fails with a confusing error

### ISSUE THREAD

**Issue by vcaputo** [solvespace/solvespace#356]
### System information

SolveSpace version: 3.0~4d1e1341

Operating system: Debian 9.5 amd64

### Expected behavior

What should have happened?

Linked the file

### Actual behavior

What actually happened?

Failed with nonsensical error.

### Additional information

If I instead go through "New Group -> Link / Assemble -> foo.slvs" it succeeds, so it's not a file-specific problem.  (the file in question doesn't reference other .slvs files either, it's a simple cylinder)

In summary:

"Link Recent" fails, showing a 'The linked file "" is not present' dialog, *twice* in quick succession.

A subsequent attempt at linking instead via "Link / Assemble" selecting the same file again shows the 'The linked file "" is not present" dialog, but only once, and succeeds at linking the file.

On a fresh start, going directly to "Link /Assemble" on the same file shows no such dialog and just works.

**Comment by whitequark** [comment]
Can you post your `~/.config/solvespace/settings.json` file here?

**Comment by vcaputo** [comment]
https://ruiner.halo.nu/~swivel/tmp/solvespace-settings.json

**Comment by whitequark** [comment]
Thanks. I can't reproduce that. I know why you might be getting two dialogs in quick succession, but I believe I've fixed all of the bugs like that a while ago, and the code seems right to me.

Can you break in gdb at `LocateImportedFile` and show me the result of `info locals`?

**Comment by vcaputo** [comment]
Ok, it looks like I made a mistake with asserting the file actually exists; it doesn't, one of the directories in the path had changed names recently.

However, there is still a bug with the dialog box showing "" for the file path, and there appears to be some kind of stuck state causing the dialog to recur even when loading the existing file.

@whitequark maybe you can reproduce what I'm seeing by renaming/reomving the file before opening it via "Link Recent"?

**Comment by whitequark** [comment]
@vcaputo Yes. I can reproduce it now.

**Comment by whitequark** [comment]
Fixed in eb7e12b.
